src/util/metadata.ts

```diff
@@ -105,10 +105,8 @@ export function createDecoratedConstants(
 	registry: TypeRegistry,
 	metadataRpc: string
 ): Constants {
-	return decorateConstants(
-		registry,
-		createMetadata(registry, metadataRpc).asLatest
-	);
+	const metadata = createMetadata(registry, metadataRpc);
+	return decorateConstants(registry, metadata.asLatest, metadata.version);
```

Comment:
Mildly confused about API here, but I reckon it's upstream. I'd have expected `.version` to also require `.asLatest` here, otherwise how do you know which version info are you grabbing? :man_shrugging:

Reply:
I think under the hood polkadot-js metadata is always the shape of the latest metadata. And assuming stuff is only added to metadata its ok because you can just have place holders in new fields when you are using it to represent old metadata. Not sure if thats exactly how it works but from reading the code thats about my understanding

Reply:
^^ Exactly this. Also I think the `.asLatest` method is misleading, per the docs it `Returns the wrapped values as a latest version object`.